lib/WebpackOptionsApply.js

```diff
@@ -321,13 +321,16 @@ class WebpackOptionsApply extends OptionsApply {
 		new RequireContextPlugin().apply(compiler);
 		new ImportPlugin().apply(compiler);
 		new SystemPlugin().apply(compiler);
-		new ImportMetaPlugin().apply(compiler);
-		new URLPlugin().apply(compiler);
-		new WorkerPlugin(
-			options.output.workerChunkLoading,
-			options.output.workerWasmLoading,
-			options.output.module
-		).apply(compiler);
+
+		if (!options.output.library || options.output.library.type !== "module") {
+			new ImportMetaPlugin().apply(compiler);
+			new URLPlugin().apply(compiler);
+			new WorkerPlugin(
+				options.output.workerChunkLoading,
+				options.output.workerWasmLoading,
+				options.output.module
+			).apply(compiler);
+		}
```

Comment:
It is breaking change, library can use `new URL(...)`, I think we should rewrite `new URL(import.meta.url)` on `new URL(__webpack_meta_url__)` and when you output ESM - get value from real `import.meta.url`, otherwise using `currentScript` for browser target, need thing about `node` target not esm

Reply:
Look at rollup output https://rollupjs.org/repl/?version=2.60.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmNvbnNvbGUubG9nKGltcG9ydC5tZXRhLnVybCklMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtYXRocy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjBzcXVhcmUlMjBmcm9tJTIwJy4lMkZzcXVhcmUuanMnJTNCJTVDbiU1Q25leHBvcnQlMjAlN0JkZWZhdWx0JTIwYXMlMjBzcXVhcmUlN0QlMjBmcm9tJTIwJy4lMkZzcXVhcmUuanMnJTNCJTVDbiU1Q25leHBvcnQlMjBmdW5jdGlvbiUyMGN1YmUlMjAoeCUyMCklMjAlN0IlNUNuJTVDdHJldHVybiUyMHNxdWFyZSh4KSUyMColMjB4JTNCJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJzcXVhcmUuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZGVmYXVsdCUyMGZ1bmN0aW9uJTIwc3F1YXJlJTIwKCUyMHglMjApJTIwJTdCJTVDbiU1Q3RyZXR1cm4lMjB4JTIwKiUyMHglM0IlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMnN5c3RlbSUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

Reply:
Why we need `__webpack_meta_url__`? Like public path for dynamic overriding..., but it can be done not here